### PR TITLE
Update MHV useDatadogRum hook

### PR DESCRIPTION
**Pull request by alex-morgun-1**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- address datadog rum defect where first initialization application configuration prevail over further initializations.
- per datadog support, need to clear _dd_s cookie to reinitialize properly

## Related issue(s)

- [MHV-64195](https://jira.devops.va.gov/browse/MHV-64195)
- https://dsva.slack.com/archives/C01G6J7UGGH/p1730227699142919
- https://dsva.slack.com/archives/CSFV4QTKN/p1730488489186209

## Testing done

- tested in DD RUM

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/platform/mhv/hooks/useDatadogRum.jsx [hunk 1]
```diff
@@ -3,7 +3,34 @@ import { datadogRum } from '@datadog/browser-rum';
 
 import { environment } from '@department-of-veterans-affairs/platform-utilities/exports';
 
+const getCookie = name => {
```
Comment (mdewey) [inline src/platform/mhv/hooks/useDatadogRum.jsx:6]: TESTING:  I think we should have some sort of unit test around this function. It seems to the core of the logic, so some sort of test would be good. Even if it's just the happy path and few sad paths

File: src/platform/mhv/hooks/useDatadogRum.jsx [hunk 2]
```diff
@@ -3,7 +3,34 @@ import { datadogRum } from '@datadog/browser-rum';
...
+  if (!ddSCookie) {
+    return;
+  }
+
+  const cookieParts = ddSCookie.split('&');
+  const idPart = cookieParts.find(part => part.startsWith('id='));
+
+  if (!idPart) {
+    // Cookie _dd_s does not have an id value. nullify the cookie to reinitialize the DD_RUM.
+    document.cookie = '_dd_s=; Path=/; Expires=Thu, 01 Jan 1970 00:00:01 GMT;';
```
Comment (mdewey) [inline src/platform/mhv/hooks/useDatadogRum.jsx:28]: QUESTION: how does this interact with other cookies? Does this delete all cookies?

**Review by mdewey — COMMENTED** [review]
Overall looks good, I have some questions and comments, but good work

File: src/platform/mhv/hooks/useDatadogRum.jsx (hunk 2 above)
Comment (oleksii-morgun) [reply src/platform/mhv/hooks/useDatadogRum.jsx:28]: no, just the `_dd_s`. it is parsed by `getCookie` function

File: src/platform/mhv/hooks/useDatadogRum.jsx (hunk 1 above)
Comment (oleksii-morgun) [reply src/platform/mhv/hooks/useDatadogRum.jsx:6]: good point. Do you recommend exporting this function from some util class?

File: src/platform/mhv/hooks/useDatadogRum.jsx (hunk 2 above)
Comment (dcloud) [reply src/platform/mhv/hooks/useDatadogRum.jsx:28]: The cookie api is wild; I had the same question, and [mdn says](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie#write_a_new_cookie):

> In the code above, newCookie is a string of form key=value, specifying the cookie to set/update. Note that you can only set/update a single cookie at a time using this method.

Comment (mdewey) [reply src/platform/mhv/hooks/useDatadogRum.jsx:28]: That is wild. Thanks!

**Review by mdewey — APPROVED** [review]
Lgtm

**Pull request merged**